scenes/player/player.gd

```diff
@@ -32,11 +32,18 @@ func _set_mode(new_mode: Mode) -> void:
 		return
 	match mode:
 		Mode.COZY:
-			player_interaction.process_mode = ProcessMode.PROCESS_MODE_INHERIT
-			player_fighting.process_mode = ProcessMode.PROCESS_MODE_DISABLED
+			_toggle_player_behavior(player_interaction, true)
+			_toggle_player_behavior(player_fighting, false)
 		Mode.FIGHTING:
-			player_interaction.process_mode = ProcessMode.PROCESS_MODE_DISABLED
-			player_fighting.process_mode = ProcessMode.PROCESS_MODE_INHERIT
+			_toggle_player_behavior(player_interaction, false)
+			_toggle_player_behavior(player_fighting, true)
+
+
+func _toggle_player_behavior(behavior_node: Node2D, is_active: bool):
+	behavior_node.visible = is_active
```

Comment:
OK, so we are tying the process mode and the visibility together.

Reply:
> OK, so we are tying the process mode and the visibility together.

Yes, at least for these 2 behaviors I thought it made sense. We can see how it evolves with more of these components.
A more future proof way would be for each of those nodes to handle its own _toggle on_ and _toggle off_, but since we only have 2 cases I wasn't sure about changing that yet.